web-common/src/components/forms/MultiInput.svelte

```diff
@@ -0,0 +1,116 @@
+
+    event.preventDefault();
+    $form[id] = $form[id].concat({ email: editingInput });
+    errs = errs.concat({ email: "" });
+    editingInput = "";
+    validateField(`${id}.${$form[id].length - 1}.${accessorKey}`);
+  }
+
+  function handleRemove(index: number) {
+    $form[id] = $form[id].filter((_, i) => i !== index);
+    errs = errs.filter?.((r, i) => i !== index);
+  }
+
+  $: hasSomeErrors = errs.some((e) => !!e[accessorKey]);
+  $: invalidValues = values.filter((_, i) => !!errs[i][accessorKey]);
+</script>
+
+<div class="flex flex-col">
+  {#if label}
+    <div class="flex items-center gap-x-1">
+      <label for={id} class="text-gray-800 text-sm font-medium">{label}</label>
+      {#if hint}
+        <Tooltip location="right" alignment="middle" distance={8}>
+          <div class="text-gray-500" style="transform:translateY(-.5px)">
+            <InfoCircle size="13px" />
+          </div>
+          <TooltipContent maxWidth="400px" slot="tooltip-content">
+            {hint}
+          </TooltipContent>
+        </Tooltip>
+      {/if}
+      {#if description}
+        <div class="text-sm text-slate-600">{description}</div>
+      {/if}
+    </div>
+  {/if}
+  <div
+    class="flex flex-wrap gap-2 bg-white rounded-sm border border-gray-300 px-3 py-[5px] w-full"
+    class:outline={focused || hasSomeErrors}
+    class:outline-red-500={hasSomeErrors}
+    class:outline-primary-500={focused && !hasSomeErrors}
+  >
+    {#each values as _, i}
+      {@const hasError = !!errs[i]?.[accessorKey]}
+      <div
+        class="flex items-center text-gray-600 text-sm rounded-2xl border border-gray-300 bg-gray-100 pl-2 pr-1 max-w-full"
+        class:border-gray-300={hasError}
+      >
+        {#if hasError}
+          <Tooltip distance={8}>
+            <div class="w-fit h-5 overflow-hidden text-ellipsis text-red-500">
+              {values[i][accessorKey]}
+            </div>
+            <TooltipContent maxWidth="400px" slot="tooltip-content">
+              {errs[i][accessorKey]}
+            </TooltipContent>
+          </Tooltip>
+        {:else}
+          <div class="w-fit h-5 overflow-hidden text-ellipsis">
+            {values[i][accessorKey]}
+          </div>
+        {/if}
+        <IconButton disableHover on:click={() => handleRemove(i)}>
+          <XIcon size="16px" className="text-gray-500 cursor-pointer" />
+        </IconButton>
+      </div>
+    {/each}
+    <input
+      bind:value={editingInput}
+      on:keydown={handleKeyDown}
+      autocomplete="off"
+      id="{id}.{values.length}.{accessorKey}"
+      class="focus:outline-white group-hover:text-red-500 text-sm w-full"
+      on:focusin={() => (focused = true)}
+      on:focusout={() => (focused = false)}
+    />
+  </div>
+  {#if hasSomeErrors}
+    <div in:slide={{ duration: 200 }} class="text-red-500 text-sm py-px">
+      Invalid {label} value{invalidValues.length > 1 ? "s" : ""}: {invalidValues
+        .map((v) => `"${v[accessorKey]}"`)
+        .join(",")}
```

Comment:
A few reactions:
- Can we use an array-wide error message from the Yup schema?
- If not, because we provide specific validation/error messages in the Yup schema, what about using an error message (the first one?) from the `errors` array?
- Currently, this message results in messy capitalization:
<img width="339" alt="image" src="https://github.com/rilldata/rill/assets/14206386/6ec3228b-5029-4ea4-b291-daf64bf92753">

Reply:
We could just not add `label` ("Emails" in this pic) ? I just thought it would be good to show all the values that are incorrect. @jkhwu Please let us know what the error message should look like.

Reply:
I think it's sufficient that the tag text turns red for all invalid values. Additionally putting the invalid values below the textbox is a bit duplicative.

Reply:
Got it. We can always tweak before the release.